apps/extension/src/routes/page/onboarding/generate.tsx

```diff
@@ -14,20 +14,74 @@ import { generateSelector } from '../../../state/seed-phrase/generate';
 import { usePageNav } from '../../../utils/navigate';
 import { PagePath } from '../paths';
 import { WordLengthToogles } from '../../../shared/containers/word-length-toogles';
+import { ChainRegistryClient } from '@penumbra-labs/registry';
+import { fetchBlockHeight } from '../../../hooks/full-sync-height';
+import { localExtStorage } from '../../../storage/local';
+import { sample } from 'lodash';
+
+const fetchBlockHeightWithFallback = async (endpoints: string[]): Promise<number> => {
+  if (endpoints.length === 0) {
+    throw new Error('All RPC endpoints failed to fetch the block height.');
+  }
+
+  // Randomly select an rpc endpoint from the chain registry
+  const randomEndpoint = sample(endpoints);
+
+  try {
+    const walletCreationBlockHeight = await fetchBlockHeight(randomEndpoint!);
+    if (walletCreationBlockHeight !== undefined) {
+      return walletCreationBlockHeight;
+    } else {
+      // Remove the current endpoint from the list and try again
+      const remainingEndpoints = endpoints.filter(endpoint => endpoint !== randomEndpoint);
+      return fetchBlockHeightWithFallback(remainingEndpoints);
+    }
+  } catch (error) {
+    const remainingEndpoints = endpoints.filter(endpoint => endpoint !== randomEndpoint);
+    return fetchBlockHeightWithFallback(remainingEndpoints);
+  }
+};
 
 export const GenerateSeedPhrase = () => {
   const navigate = usePageNav();
   const { phrase, generateRandomSeedPhrase } = useStore(generateSelector);
   const [count, { startCountdown }] = useCountdown({ countStart: 3 });
   const [reveal, setReveal] = useState(false);
+  const [blockHeight, setBlockHeight] = useState<number | null>(null);
+
+  // On render, generate a new seed phrase.
+  // Fetch data only once on the initial render.
+  const isFetchedRef = useRef(false);
 
-  // On render, generate a new seed phrase
   useEffect(() => {
+    const fetchData = async () => {
+      if (isFetchedRef.current) {
+        return;
+      }
+
+      try {
+        const chainRegistryClient = new ChainRegistryClient();
+        const { rpcs } = chainRegistryClient.bundled.globals();
+
+        const suggestedEndpoints = rpcs.map(i => i.url);
+        const blockHeight = await fetchBlockHeightWithFallback(suggestedEndpoints);
+        await localExtStorage.set('walletCreationBlockHeight', blockHeight);
+        setBlockHeight(blockHeight);
+
+        isFetchedRef.current = true;
+      } catch (error) {
+        console.error('Error fetching data:', error);
+      }
+    };
+
```

Comment:
issue: think we should move all this logic to the SeedPhraseSlice or somewhere in zustand (fetchBlockHeightWithFallback too)

Reply:
this was a great opportunity for me to refresh my knowledge about Zustand. I moved the suggested logic to a dedicated zustand store called `walletCreationBlockHeightSlice`, and `fetchBlockHeightWithFallback` and `fetchBlockHeight` are now utility functions.

Reply:
modified to use react query pattern